lib/archethic/transaction_chain/transaction/data/recipient/arguments_encoding.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Archethic.TransactionChain.TransactionData.Recipient.ArgumentsEncoding do
+  @moduledoc """
+  Handle encoding of recipients arguments
+  """
+
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.Utils.TypedEncoding
+
+  @spec serialize(args :: list(TypedEncoding.arg()), mode :: Transaction.serialization_mode()) ::
+          bitstring()
+  def serialize(args, mode) when is_list(args) and mode in [:compact, :extended] do
+    bit_size =
+      case mode do
+        :compact ->
+          1
+
+        :extended ->
+          8
+      end
+
+    bin =
+      args
+      |> Enum.map(&TypedEncoding.serialize(&1, bit_size))
+      |> :erlang.list_to_bitstring()
+
+    <<length(args)::8, bin::bitstring>>
```

Comment:
**Suggestion**
Encoding the length of the args list on 8 bits means the list could have a length of 255 items maximum.
This limitation should also be added in the json schema for the JSON RPC APIs